src/DotNetLightning.Core/Transactions/Transactions.fs

```diff
@@ -735,20 +772,31 @@ module Transactions =
                 else
                     spec.ToLocal.ToMoney(), spec.ToRemote.ToMoney() - closingFee
 
-            let maybeToLocalOutput = if (toLocalAmount >= dustLimit) then Some(toLocalAmount, localDestination) else None
-            let maybeToRemoteOutput = if (toRemoteAmount >= dustLimit) then Some(toRemoteAmount, remoteDestination) else None
-            let outputs = seq [maybeToLocalOutput; maybeToRemoteOutput] |> Seq.choose id
+            let outputs =
+                seq {
+                    if toLocalAmount >= dustLimit then
+                        yield (toLocalAmount, localDestination)
+                    if toRemoteAmount >= dustLimit then
+                        yield (toRemoteAmount, remoteDestination)
+                }
+                |> Seq.sortBy (fun (money, dest) ->
+                    let txOut = TxOut(money, dest)
+                    use memoryStream = new MemoryStream()
+                    let stream = NBitcoin.BitcoinStream(memoryStream, true)
+                    txOut.ReadWrite stream
+                    memoryStream.ToArray()
```

Comment:
Isn't this same with `txOut.ToBytes()` ?

Reply:
I don't know, I copied this bit from @canndrew's PR. If it is, you prefer ToBytes()?

Reply:
Yes, `ReadWrite` is very bitcoin-specific and the reader have to know about the context to understand. So I prefer `ToBytes`